test/src/tasks/publish-task.test.ts

```diff
@@ -0,0 +1,99 @@
+import { assert } from 'chai'
+import { useEnvironment } from '~/test/test-helpers/useEnvironment'
+import { TASK_PUBLISH } from '~/src/tasks/task-names'
+import { AragonConfig } from '~/src/types'
+import { startGanache } from '~/src/tasks/start/backend/start-ganache'
+import deployBases from '~/src/tasks/start/backend/bases/deploy-bases'
+import {
+  PublishVersionTxData,
+  resolveRepoContentUri,
+  isAddress
+} from '~/src/utils/apm'
+import { infuraIpfsApiUrl, infuraIpfsGateway } from '~/test/testParams'
+import { defaultLocalAragonBases } from '~/src/params'
+
+const testAppDir = 'test-app'
+const testAppContract = 'TestContract'
+const testNetwork = 'localhost'
+
+describe(`Run publish task - ${testAppDir}`, function() {
+  const ipfsApiUrl = infuraIpfsApiUrl
+  const ipfsGateway = infuraIpfsGateway
+  const managerAddress = '0xAb5801a7D398351b8bE11C439e05C5B3259aeC9B'
+  const bump = 'major'
```

Comment:
It would be nice to have these three cases covered:

1. Repo doesn't exist; deploys new contract and repo
2. Repo exists, do a minor or patch with just content changing
3. Repo exists, do a major with new contract and content

Right now, I think we're only covering 1?

Reply:
We should definitely cover all 3 cases. Should I do that in this PR or in a subsequent PR?

Reply:
Let's do so in a different PR since this one is already getting big :).

Reply:
Added to #77